### PR TITLE
README: Android Studio workarounds

### PR DESCRIPTION
**Pull request by ruslo**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -268,6 +268,54 @@ There is another entry point for Android Studio - ``src/examples/facefilter/andr
 It should be used only for testing or as a template for starting your own project
 based on Drishti.
 
+Android Studio workarounds
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Next factors add an instability to the Android Studio build:
```
Comment (headupinclouds) [inline README.rst]: The following factors can all contribute to some instability in the Android Studio managed build:

File: README.rst [hunk 2]
```diff
@@ -268,6 +268,54 @@ There is another entry point for Android Studio - ``src/examples/facefilter/andr
...
+
+* Android Studio 3.2 is **beta**
+* Using custom CMake 3.7+ in Android Studio is `a preview feature <https://developer.android.com/studio/projects/add-native-code#vanilla_cmake>`__
+* Some issues are hard to track or confirm, some issues `like this one <https://issuetracker.google.com/issues/75268076>`__ is already reported but still **not fixed**
```
Comment (headupinclouds) [inline README.rst]: some issues [link] >>are<< already reported

File: README.rst [hunk 3]
```diff
@@ -268,6 +268,54 @@ There is another entry point for Android Studio - ``src/examples/facefilter/andr
...
+
+From experience the weakest part in build is a communication between Gradle and
```
Comment (headupinclouds) [inline README.rst]: From experience >>,<< the weakest part in >>the<< build is >>the<< communication between Gradle and CMake>>.<<  To minimize it, the >>following<< trick can be used:

File: README.rst [hunk 4]
```diff
@@ -268,6 +268,54 @@ There is another entry point for Android Studio - ``src/examples/facefilter/andr
...
+CMake, to minimize it the next trick can be used:
+
+* Open top-level ``CMakeLists.txt`` file
```
Comment (headupinclouds) [inline README.rst]: Open >>the<< top-level

File: README.rst [hunk 5]
```diff
@@ -268,6 +268,54 @@ There is another entry point for Android Studio - ``src/examples/facefilter/andr
...
+If you're running it a first time there will be a high chance to hit this
+Gradle issue:
+
+.. code-block:: none
+
+  * What went wrong:
+  Execution failed for task '...'.
+  > Conversion = c, Flags =
+
+In this case just wait for few seconds and run Gradle again:
```
Comment (headupinclouds) [inline README.rst]: In this case>>,<< just wait for

File: README.rst [hunk 6]
```diff
@@ -268,6 +268,54 @@ There is another entry point for Android Studio - ``src/examples/facefilter/andr
...
+
+.. code-block:: none
+
+  [drishti/android-studio]> ./gradlew assembleDebug
+
+* Revert ``CMakeLists.txt`` file, i.e. substitute ``if(TRUE)`` with ``if(DRISHTI_DEBUG_STOP)``.
+
+* Run CMake build without Gradle:
```
Comment (headupinclouds) [inline README.rst]: Run >>the<< CMake build without

File: README.rst [hunk 7]
```diff
@@ -268,6 +268,54 @@ There is another entry point for Android Studio - ``src/examples/facefilter/andr
...
+
+.. code-block:: none
+
+  [drishti/android-studio]> cmake --build ../src/examples/facefilter/android-studio/app/.externalNativeBuild/cmake/debug/arm64-v8a
+
+Once CMake build is ready you can use ``./gradlew assembleDebug`` or open
```
Comment (headupinclouds) [inline README.rst]: Once >>the<< CMake build is ready>>,<< you can use

Comment (headupinclouds) [reply README.rst]: While this is less than ideal, the builds have been manageable in practice, and the **beta** issues should be resolved over time.

**Comment by ruslo** [comment]
> Please see grammatical touch ups

Thanks, fixed.

**Pull request merged**